courses/management/commands/generate_certificate.py

```diff
@@ -0,0 +1,65 @@
+"""
+Management command to generate certificates
+"""
+from django.core.management.base import BaseCommand, CommandError
+from courses.utils import sync_courserun_grades, sync_courserun_certificates
+
+
+class Command(BaseCommand):
+    """
+    Command to generate certificates for a course.
+    """
+
+    help = "Generates certificates for a course for all enrolled users or a specified user."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--email",
+            type=str,
+            help="The email address of the enrolled User.",
+            required=False,
+        )
+        parser.add_argument(
+            "--course-run",
+            type=str,
+            help="The 'courseware_id' value for a CourseRun.",
+            required=True,
+        )
+        parser.add_argument(
+            "--grade",
+            type=float,
+            help="Override a grade. Setting grade: 0 blacklists a user. Setting a passing grade \
+                (>0.0) whitelists a user. range: 0.0 - 1.0",
+            required=False,
+        )
+        parser.add_argument(
+            "--force",
+            action="store_true",
+            help="Force overwrite of local grade record.",
+        )
+        super().add_arguments(parser)
+
+    def handle(self, *args, **options):
+        """Handle command execution"""
+        email = options["email"]
+        readable_id = options["course_run"]
+        grade = float(options["grade"])
+        force = options["force"]
+
+        # Grade override for all users for the course run. Disallowed.
+        if grade and not email:
+            raise CommandError(
+                "No user supplied with override grade. Force overwrite of grade is not supported \
+                    for all users. Grade should only be supplied when a specific user is targeted."
+            )
+
+        if grade < 0.0 or grade > 1.0:
+            raise CommandError("Invalid value for grade. Allowed range: 0.0 - 1.0")
```

Comment:
I was thinking some more about this "grade override" and I realized that the PM will more often need to override the passing status, not the grade. The most common case will probably be making no change to the grade, but overriding whether the user passed or failed.

Reply:
@pdpinch Right now the command is setup to use the `grade` value is passing/failing. We can indeed switch to the `passed` attribute value as well. I'll need some clear specification of logic for this. Most of the stuff is already in there, might need just a few small tweaks.

Reply:
@pdpinch another issue that we have with only using the `passed` status is how to indicate a blacklist.